pilot/pkg/model/authorization.go

```diff
@@ -191,6 +196,23 @@ func (policy *AuthorizationPolicies) RoleToBindingsForNamespace(ns string) map[s
 	return rolesAndBindings.RoleNameToBindings
 }
 
+// RoleForNameAndNamespace returns a ServiceRole from this namespace, given the ServiceRole name.
+// This function always return a non nil struct instance.
+func (policy *AuthorizationPolicies) RoleForNameAndNamespace(ns, roleName string) *rbacproto.ServiceRole {
+	if policy == nil || policy.NamespaceToAuthorizationConfigV2 == nil {
+		return &rbacproto.ServiceRole{}
+	}
+	nsToAuthzConfigV2 := policy.NamespaceToAuthorizationConfigV2[ns]
+	if nsToAuthzConfigV2 == nil {
+		return &rbacproto.ServiceRole{}
+	}
+	serviceRole, exist := policy.NamespaceToAuthorizationConfigV2[ns].NameToServiceRoles[roleName]
```

Comment:
Not use `nsToAuthzConfigV2`?

Reply:
I don't understand. Could you be more specific? Thanks!

Reply:
You have the variable `nsToAuthzConfigV2` in line 205, why not use it here?

Reply:
Right. Thanks :D